docs/index.asciidoc

```diff
@@ -0,0 +1,425 @@
+are environments, users and products. Locally there is a database
+that the enhancement lookups are done against. Remotely there is
+a database that has the static data. The local db is loaded with
+data from the remote db once or very periodically. To achieve this
+three main sections are defined: db_objects, loaders and lookups.
+The local database is an Apache Derby in-memory database.
+
+Db_objects
+
+Define as many of theses as needed to build the local database
+structure. Define tables, columns and indexes.
+
+Loaders
+
+Define as many loaders as needed to fetch the remote data. Each
+loader should fill a table created via a db_object. Make sure
+the column names/datatypes in the loader SQL statement match the
+columns defined in its corresponding db_object. Each loader has
+an independent remote database connection.
+
+Lookups
+
+Define as many lookups as needed to enhance the event from all
+lookup tables in one pass. Ideally the SQL statement should only
+return one row. Any rows are converted to Hash objects and are
+stored in a target field that is an Array.
+
+Example config:
+
+[source,ruby]
+filter {
+  jdbc_static {
+    loaders => [
+      {
+        id => "remote-servers"
+        query => "select ip, descr from ref.local_ips order by ip"
+        local_table => "servers"
+      },
+      {
+        id => "remote-users"
+        query => "select firstname, lastname, userid from ref.local_users order by userid"
+        local_table => "users"
+      }
+    ]
+    local_db_objects => [
+      {
+        name => "servers"
+        index_columns => ["ip"]
+        columns => [
+          ["ip", "varchar(15)"],
+          ["descr", "varchar(255)"]
+        ]
+      },
+      {
+        name => "users"
+        index_columns => ["userid"]
+        columns => [
+          ["firstname", "varchar(255)"],
+          ["lastname", "varchar(255)"],
+          ["userid", "int"]
+        ]
+      }
+    ]
+    local_lookups => [
+      {
+        id = "local-servers"
+        query => "select descr as description from servers WHERE ip = :ip"
+        parameters => {ip => "[from_ip]"}
+        target => "server"
+      },
+      {
+        id = "local-users"
+        query => "select firstname, lastname from users WHERE userid = :id"
+        parameters => {id => "[loggedin_userid]"}
+        target => "user"
+      }
+    ]
+    # using add_field here to add & rename values to the event root
+    add_field => { server_name => "%{[server][0][description]}" }
+    add_field => { user_firstname => "%{[user][0][firstname]}" }
+    add_field => { user_lastname => "%{[user][0][lastname]}" }
+    remove_field => ["server", "user"]
+    jdbc_user => "logstash"
+    jdbc_password => "example"
+    jdbc_driver_class => "org.postgresql.Driver"
+    jdbc_driver_library => "/tmp/logstash/vendor/postgresql-42.1.4.jar"
+    jdbc_connection_string => "jdbc:postgresql://remotedb:5432/ls_test_2"
+  }
+}
+
```

Comment:
I wonder if an example here - with tables and results - along the lines of the [one you had in your PR](https://github.com/logstash-plugins/logstash-filter-jdbc_static/pull/2#issuecomment-348324373) would be useful to users?

Reply:
I can add that.

Reply:
Done below.